client/engine/engine.go

```diff
@@ -389,10 +391,13 @@ func (e *Engine) constructObjectiveFromMessage(id protocols.ObjectiveId, ss stat
 		}
 		return &vfo, nil
 	case directdefund.IsDirectDefundObjective(id):
-		ddfo, err := directdefund.ConstructObjectiveFromState(ss.State(), e.store.GetChannelById)
+		ddfo, err := directdefund.ConstructObjectiveFromState(ss.State(), e.store.GetConsensusChannelById)
 		if err != nil {
 			return &directdefund.Objective{}, fmt.Errorf("could not create direct defund objective from message: %w", err)
 		}
+
+		// TODO Destroy / Disable / Disown the associated ConsensusChannel
```

Comment:
This should become a new issue or be tracked in #598

Reply:
Created #607